apps/dashboard/pages/api/v1/github/verify.ts

```diff
@@ -0,0 +1,196 @@
+  const gitHub = await fetch(GITHUB_KEYS_URI);
+  if (!gitHub.ok) {
+    throw new Error("No public keys found");
+  }
+  const gitBody = gitHub.ok ? await gitHub.json() : undefined;
+  //Get public keys from Github
+  const public_key =
+    gitBody.public_keys.find((k: Key) => {
+      if (k.key_identifier === keyID) {
+        return k;
+      }
+    }) ?? null;
+
+  if (!public_key) {
+    console.error("No public keys found");
+  }
+  // Verify signature
+  const verify = crypto.createVerify("SHA256").update(payload);
+  if (!verify.verify(public_key.key, Buffer.from(signature.toString(), "base64"))) {
+    return false;
+  }
+  return true;
+};
+
+export default async function handler(request: NextApiRequest, response: NextApiResponse) {
+  const signature = request.headers["github-public-key-signature"]?.toString();
+  const keyId = request.headers["github-public-key-identifier"]?.toString();
+  async function getRawBody(readable: Readable): Promise<Buffer> {
+    const chunks = [];
+    for await (const chunk of readable) {
+      chunks.push(typeof chunk === "string" ? Buffer.from(chunk) : chunk);
+    }
+    return Buffer.concat(chunks);
+  }
+
+  const rawBody = await getRawBody(request);
+
+  const data = JSON.parse(Buffer.from(rawBody).toString("utf8"));
+
+  if (!signature) {
+    throw new Error("No signature found");
+  }
+  if (!keyId) {
+    throw new Error("No KeyID found");
+  }
+  if (data?.length === 0) {
+    throw new Error("No data found");
+  }
+  // Worked but needs to be off while other functions are tested. Hashed with example from site so will always fail till live
+  const isGithubVerified = await verify_git_signature(rawBody.toString(), signature, keyId);
+  if (!isGithubVerified) {
+    throw new Error("Github signature not verified");
+  }
+
+  for (const item of data) {
+    const token = item.token.toString();
+    const hashedToken = await sha256(token);
+    const isKeysFound = await db.query.keys.findFirst({
+      where: (table, { and, eq, isNull }) =>
+        and(eq(table.hash, hashedToken), isNull(table.deletedAt)),
+    });
+    const ws = await db.query.workspaces.findFirst({
+      where: (table, { and, eq, isNull }) =>
+        and(
+          eq(table.id, isKeysFound?.forWorkspaceId ? isKeysFound?.forWorkspaceId : ""),
+          isNull(table.deletedAt),
+        ),
+    });
+    if (!ws) {
+      throw new Error("workspace does not exist");
+    }
+    const users = await getUsers(ws.tenantId);
+    const date = new Date().toDateString();
+    for await (const user of users) {
+      await resend.sendLeakedKeyEmail({
+        email: user.email,
+        date: date,
+        source: item.source,
+        type: item.type,
+        url: item.url,
+      });
+      const text1 = `Type: ${item.type} \n Source: ${item.source} \n Date: ${date} \n URL: ${item.url}`;
+      const text2 = `Key: ${isKeysFound?.id} \n Workspace: ${ws.name} \n Tenant: ${ws.tenantId} \n User: ${user.email}`;
```

Comment:
I think it makes more sense to just item.type, source date, id and name and tenant id and email as props passed to the slack handler versus have the text passed to the function.